src/base/views/viewer.tsx

```diff
@@ -59,19 +60,21 @@ export class ModelRenderer implements RenderingContext {
             vnode);
     }
 
-    renderElement(element: Readonly<SModelElement>, args?: object): VNode | undefined {
+    renderElement(element: Readonly<SModelElement>, args?: IViewArgs): VNode | undefined {
         const view = this.viewRegistry.get(element.type);
-        const vnode = view.render(element, this, args);
+        const vnode = view.render(element, this, {...args, ...this.args});
         if (vnode) {
             return this.decorate(vnode, element);
         } else {
             return undefined;
         }
     }
 
-    renderChildren(element: Readonly<SParentElement>, args?: object): VNode[] {
+    renderChildren(element: Readonly<SParentElement>, args?: IViewArgs): VNode[] {
+        const childArgs = args ? { ...args, parentArgs: this.args } : this.args;
+        const context = new ModelRenderer(this.viewRegistry, this.targetKind, this.postprocessors, childArgs);
```

Comment:
To avoid creating unnecessary new instances of ModelRenderer, we should reuse `this` if `args` is undefined.

Reply:
Right, I appended a commit that avoids creating new objects if `args` is `undefined`.